bin/client/main.rs

```diff
@@ -26,7 +26,24 @@ mod opts;
 use opts::*;
 
 #[tokio::main]
-async fn main() -> std::result::Result<(), ClientError> {
+async fn main() {
+    // Trap and format error messages using the proper value
+    if let Err(e) = run().await.map_err(|e| anyhow::Error::new(e)) {
+        eprintln!("{}", e);
+        for (i, cause) in e.chain().enumerate() {
+            // Skip the first message
```

Comment:
Nit: I forgot to add to this to say we skip it because it is the same as the initial error that gets printed

Reply:
Fixed